src/hooks/useDataSource.ts

```diff
@@ -107,12 +108,16 @@ export default function useDataSource(): DataSourceContextType {
 
 async function save(storage: BookStorage) {
   DATASOURCE.options.extra.queryClient.setQueryData(['state', 'isSaving'], true);
+  await DATASOURCE.query('VACUUM');
   const rawBook = DATASOURCE.sqljsManager.exportDatabase();
   await storage.save(rawBook);
   DATASOURCE.options.extra.queryClient.setQueryData(['state', 'isSaving'], false);
 }
 
 async function importBook(storage: BookStorage, rawData: Uint8Array) {
+  if (isStaging()) {
+    Settings.now = () => Date.now();
+  }
```

Comment:
unrelated but this allows users to import their own files and go to any date. Without this, can only check their data up until 1st Jan 2023